### PR TITLE
Pseudo randomly add checks

### LINKED ISSUE #7127: [Testing] Implement abiv2 testing with holes for efficiency
## Abstract

At the moment, the abiv2 proto converter initializes all array elements and resizes every dynamic dimension.

For larger dimensional arrays, this leads to large test programs some of which take several minutes to execute on evmone.

This issue tracks support in the abiv2 converter for pseudo randomly resizing dynamic array elements and initializing array indices. We gain efficiency at the cost an inextensive check of abiv2 coder.

The hope is that this will still find issues eventually. Arguably, it is better to run quick inextensive tests than potentially exponentially slow extensive tests.

Simple example:

```
// Current version
int8 x[3];
x[0] = 1;
x[1] = 2;
x[2] = 3;

// checks
if (x[0] != 1) return 1;
if (x[1] != 2) return 2;
if (x[2] != 3) return 3;
if (x.length != 3) return 4;

// Future version
// saves four statements but performs inexhaustive check.
x[2] = 3;
if (x[2] != 3) return 1;
if (x.length != 3) return 2;
```

## Motivation

Pseudo-randomly initializing array elements is a simpler problem for which something like the following may be proposed.

We define a constant `k` and define an array index only if the current value of the monotonic counter in the converter is divisble by `k` (i.e., `counter mod k == 0`)

I don't have a solution for pseudo-random resize of dynamic arrays. One issue I am pondering over is how to pseudo randomly resize and initialize and array like this

```
// Dimensions 1,2,4 are dynamic
// Dimensions 3,5 are static
int8 x[][][2][][10];
```

TBD

### PR DESCRIPTION
**Pull request by bshastry**
closes #7127

### CODE REVIEW

File: test/tools/ossfuzz/protoToAbiV2.h [hunk 1]
```diff
@@ -334,14 +334,27 @@ class ProtoConverter
...
-		return std::make_pair(_x.is_static(), getArrayLengthFromFuzz(_x.length()));
+		return std::make_pair(
+			_x.is_static(),
+			getStaticArrayLengthFromFuzz(_x.length())
+		);
+	}
+
+	static bool addCheck(unsigned _counter)
+	{
+		return (_counter % s_arrayCheckFrequency == 0);
```
Comment (Marenz) [inline test/tools/ossfuzz/protoToAbiV2.h:357]: no braces needed here

Comment (Marenz) [reply test/tools/ossfuzz/protoToAbiV2.h:357]: except maybe around the modular op, not sure about the precedence.

Comment (bshastry) [reply test/tools/ossfuzz/protoToAbiV2.h:357]: thanks, this always confuses me :)

**Comment by bshastry** [comment]
Closing this PR until evidence that fuzzing with holes is better. At the moment, it simply has poorer coverage than fuzzing without holes.

**Pull request closed without merging**